src/components/Categories/LoanSpotlight.vue

```diff
@@ -0,0 +1,255 @@
+							Today's loan spotlight
+						</h2>
+						<h3 class="tw-pt-2 tw-mb-1">
+							{{ getSpotlightLoanLocation }}
+						</h3>
+						<span class="tw-line-clamp-5">
+							<p
+								v-for="(paragraph, index) in getSpotlightText"
+								:key="index"
+								v-html="paragraph"
+							>
+							</p>
+						</span>
+						<div class="tw-mt-2">
+							<kv-button
+								class="tw-w-full md:tw-w-auto"
+								:to="`/lend-beta/${this.spotlightLoan.id}`"
+								variant="primary"
+								state=""
+								v-kv-track-event="['Lending', 'click-loan-spotlight', 'View Loan']"
+							>
+								View loan
+							</kv-button>
+						</div>
+					</div>
+				</div>
+			</kv-grid>
+		</kv-page-container>
+	</div>
+</template>
+
+<script>
+
+import { toParagraphs } from '@/util/loanUtils';
+import gql from 'graphql-tag';
+import logReadQueryError from '@/util/logReadQueryError';
+import KvResponsiveImage from '@/components/Kv/KvResponsiveImage';
+import KvGrid from '~/@kiva/kv-components/vue/KvGrid';
+import KvPageContainer from '~/@kiva/kv-components/vue/KvPageContainer';
+import KvButton from '~/@kiva/kv-components/vue/KvButton';
+
+const routePath = 'recommended-by-lenders';
+
+const allChannelsQuery = gql`
+	query allChannelsQuery {
+		lend {
+			loanChannels(offset:0, limit:1000) {
+				values {
+					id
+					url
+					name
+					loans {
+						totalCount
+					}
+				}
+			}
+		}
+	}
+`;
+
+const spotlightLoanQuery = gql`
+	query spotlightLoanQuery (
+		$ids: [Int],
+		$limit: Int = 5,
+		$offset: Int = 0,
+		$imgDefaultSize: String = "w520h390",
+		$imgRetinaSize: String = "w1040h780",
+	) {
+		lend {
+			loanChannelsById (ids: $ids) {
+				id
+				loans (
+					limit: $limit
+					offset: $offset
+				) {
+					values {
+						id
+						description
+						lenderRepaymentTerm
+						anonymizationLevel
+						geocode {
+							city
+							country {
+								name
+							}
+						}
+						image {
+							id
+							default: url(customSize: $imgDefaultSize)
+							retina: url(customSize: $imgRetinaSize)
+						}
+					}
+				}
+			}
+		}
+	}
+`;
+
+function getTargetedChannel(targetedRoute, allChannels) {
+	const loanChannels = allChannels.lend?.loanChannels?.values;
+	const targetedLoanChannel = loanChannels.filter(
+		loanChannel => loanChannel.url.split('/').pop() === targetedRoute
+	);
+	// no channel that matches the name recommended-by-lenders
+	if (targetedLoanChannel.length === 0) {
+		// return id for channel women
+		return 5;
+	}
+	// recommended-by-lenders channel exists but no loans exist within it
+	if (targetedLoanChannel.length !== 0 && targetedLoanChannel[0].loans.totalCount === 0) {
+		return 5;
+	}
+	// isolate recommended-by-lenders channel id
+	return targetedLoanChannel[0]?.id || null;
+}
+
+function filterByAnonymizationLevel(spotlightData) {
+	const firstFiveRecommendedLoans = spotlightData.lend?.loanChannelsById[0]?.loans?.values ?? [];
+	const nonAnonymousLoans = firstFiveRecommendedLoans.filter(
```

Comment:
Same comment about probably wanting optional chaining here.

Reply:
The nullish coalescing above means this  won't return undefined

Reply:
Ah yup, I missed the `[]`.